test/test.js

```diff
@@ -14,290 +19,476 @@ function log () {
   console.log.apply(console, args)
 }
 
-var testfile = __dirname + '/testfile.txt'
+var testfilePath = __dirname + '/testfile.txt'
+var testfile = fs.readFileSync(__dirname + '/testfile.txt')
+
+describe('IPFS Node.js API wrapper tests', function () {
+  var apiClients = {} // a, b, c
+  var ipfsNodes = {} // a, b, c
 
-describe('ipfs node api', function () {
-  var ipfs, ipfsNode
   before(function (done) {
     this.timeout(20000)
     log('ipfs node setup')
+    var counter = 0
+
+    if (isNode) {
+      startIndependentNode(ipfsNodes, apiClients, 'a', finish)
+      startIndependentNode(ipfsNodes, apiClients, 'b', finish)
+      startIndependentNode(ipfsNodes, apiClients, 'c', finish)
```

Comment:
For dynamic values in the karma.conf you can just pass those through the api call, but to ensure the tests in node as well in the browser have the right apiAddrs I suggest writing them to a file like `test/fixtures/addrs.txt` and then reading this file from the tests.